src/steamship/agents/llms/steamship_llm.py

```diff
@@ -0,0 +1,110 @@
+import logging
+from pprint import pformat
+from typing import List, Optional
+
+from steamship import Block, File, MimeTypes, PluginInstance, Steamship, SteamshipError, Tag
+from steamship.agents.schema import Tool
+from steamship.data import GenerationTag, TagKind
+from steamship.plugin.capabilities import (
+    CapabilityPluginRequest,
+    CapabilityPluginResponse,
+    ConversationSupport,
+    FunctionCallingSupport,
+    SystemPromptSupport,
+)
+
+
+class SteamshipLLM:
+    """A class wrapping LLM plugins."""
+
+    def __init__(self, plugin_instance: PluginInstance):
+        # TODO (PR callout): In the spirit of getting folks to pin to specific instances more often, take the route of
+        #  using specific PluginInstances rather than client.use_plugin all the time with the params that we determine?
+        self.client = plugin_instance.client
+        self.plugin_instance = plugin_instance
+
+    @staticmethod
+    def with_gpt4(client: Steamship, temperature: float = 0.4, max_tokens: int = 256):
```

Comment:
This is a style choice, but our `gpt-4` plugin allows other models. I think we either need to make a different builder-style method, expose `model_name` as an explicit parameter, or something equivalent.

This might also be a good time to think about renaming this to reflect provider vs. model type (or some other aspect).  `SteamshipLLM.using_openai_chatcomplete` perhaps, or (less-clear) `SteamshipLLM.using_gpt` ?